### PR TITLE
setup ossf scorecard and codql workflows

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

**Comment by mmorel-35** [comment]
cc @phlax

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: "CodeQL"
+
+permissions:
+  actions: read
```
Comment (phlax) [inline .github/workflows/codeql.yml:4]: i think just `contents: read` here and then specify in jobs as a convention

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  contents: read
+  security-events: write
+
+on:
+  push:
+    branches: [ "main" ]
```
Comment (phlax) [inline .github/workflows/codeql.yml:10]: nit: yaml

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: '41 21 * * 0'
```
Comment (phlax) [inline .github/workflows/codeql.yml:14]: nit: indents

File: .github/workflows/scorecard.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
+name: Scorecard supply-chain security
+
+permissions: read-all
```
Comment (phlax) [inline .github/workflows/scorecard.yml:3]: same here

**Review by phlax — COMMENTED** [review]
a few format nits but otherwise lgtm

**Comment by phlax** [comment]
cc @tedjpoole

**Review by tedjpoole — APPROVED** [review]

**Pull request merged**